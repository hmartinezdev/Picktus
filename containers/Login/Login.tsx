import React, { Component } from 'react';
import Logo from '@components/Logo';
import Background from '@components/Background';

class Login extends Component {
  render() {
    return (
      <Background>
        <div className="container">
          <div className="login">
            <Logo />
          </div>
          <style jsx>{`
            .container {
              width: 100%;
              height: 100vh;
              display: flex;
              justify-content: center;
              align-items: flex-start;
              padding: 2rem;
              box-sizing: border-box;
            }

            .login {
              background-color: #2d3047cd;
              padding-right: 2rem;
              box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
            }
          `}</style>
        </div>
      </Background>
    );
  }
}

export default Login;
