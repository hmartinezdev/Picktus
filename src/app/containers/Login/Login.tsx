import Background from '@components/Background';
import LoginHandler from '@components/LoginHandler';
import Logo from '@components/Logo';
import colors from '@constants/colors';
import React, { Component } from 'react';

class Login extends Component {
  public render(): React.ReactElement<Login> {
    return (
      <Background>
        <div className="container">
          <div className="login">
            <Logo />
          </div>
          <div className="form">
            <LoginHandler />
          </div>
          <style jsx>{`
            .container {
              width: 100%;
              height: 100vh;
              display: flex;
              align-items: center;
              flex-direction: column;
              padding: 2rem 0;
              box-sizing: border-box;
              overflow: auto;
            }

            .form {
              height: 100%;
              width: 100%;
              display: flex;
              flex-direction: column;
              justify-content: flex-start;
              align-items: center;
              margin-top: 2.5rem;
            }

            .login {
              background-color: ${colors.primary}cd;
              box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
              display: flex;
              position: relative;
              border-radius: 3px;
            }

            .login::after {
              position: absolute;
              top: 50%;
              left: 50%;
              transform: translate(-50%, -50%);
              content: '';
              width: calc(100% + 1.2rem);
              height: calc(100% + 1.2rem);
              border: 2px solid transparent;
              border-image: linear-gradient(to bottom, ${colors.primary} 60%, ${colors.secondary} 100%);
              border-image-slice: 1;
            }
          `}</style>
        </div>
      </Background>
    );
  }
}

export default Login;
