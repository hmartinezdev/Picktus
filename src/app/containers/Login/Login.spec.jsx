import { shallow } from 'enzyme';
import React from 'react';
import Login from './Login';

let wrapper;

const setup = () =>
  shallow(
    <Login>
      <div />
    </Login>
  );

describe('<Login />', () => {
  test('should render properly', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
