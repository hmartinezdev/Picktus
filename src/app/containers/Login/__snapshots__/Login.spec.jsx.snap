// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render properly 1`] = `
<div
  className="container container--undefined"
>
  <div
    className="login"
  >
    <Logo />
  </div>
  <div
    className="form"
  >
    <Connect(LoginHandler) />
  </div>
  <style
    jsx={true}
  >
    
          .container {
            width: 100%;
            height: 100vh;
            display: flex;
            align-items: center;
            flex-direction: column;
            padding: 2rem 0;
            box-sizing: border-box;
            overflow: auto;
            opacity: 1;
            transition: opacity 300ms ease-out;
          }

          .container--exiting,
          .container--exited {
            position: absolute;
            opacity: 0;
          }

          .form {
            height: 100%;
            width: 100%;
            display: flex;
            flex-direction: column;
            justify-content: flex-start;
            align-items: center;
            margin-top: 2.5rem;
            opacity: 1;
            transition: opacity 130ms ease-out;
          }

          .login {
            background-color: #2d3047;
            box-shadow: 11px 10px 34px -7px rgba(0, 0, 0, 0.49);
            display: flex;
            position: relative;
            border-radius: 3px;
            transition: all 300ms ease-out;
            min-height: 8.5rem;
            max-height: 8.5rem;
          }
        
  </style>
</div>
`;
